Enhancing the dbt versioning doc (#8800)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-versions/about-versions.md

<Card

>     title="About dbt Core versions"

Reviewer: ```suggestion
    title="About dbt versions"
```

>     body="Information about how dbt Core is versioned and how to target those versions."

Reviewer: ```suggestion
    body="Learn about semantic versioning for dbt engines, and how long those versions are supported."
```

<Card
    title="dbt version upgrade guides"
    body="All the information you need to prepare your projects for the next version of dbt, including Fusion. Includes guidance on new features, behavior changes, deprecations, and much more."

>     link="/docs/dbt-versions/dbt-versions-upgrade"

Author: ```suggestion
    link="/docs/dbt-versions/core-upgrade"
```
